sdks/python/apache_beam/runners/interactive/pipeline_instrument.py

```diff
@@ -65,25 +78,37 @@ def __init__(self, pipeline, options=None):
         pipeline.to_runner_api(use_fake_coders=True),
         pipeline.runner,
         options)
+
+    self._background_caching_pipeline = beam.pipeline.Pipeline.from_runner_api(
+        pipeline.to_runner_api(use_fake_coders=True),
+        pipeline.runner,
+        options)
+
     # Snapshot of original pipeline information.
     (self._original_pipeline_proto,
      self._original_context) = self._pipeline_snap.to_runner_api(
          return_context=True, use_fake_coders=True)
 
     # All compute-once-against-original-pipeline fields.
-    self._has_unbounded_source = has_unbounded_source(self._pipeline_snap)
+    self._unbounded_sources = unbounded_sources(
+        self._background_caching_pipeline)
     # TODO(BEAM-7760): once cache scope changed, this is not needed to manage
     # relationships across pipelines, runners, and jobs.
     self._pcolls_to_pcoll_id = pcolls_to_pcoll_id(self._pipeline_snap,
                                                   self._original_context)
 
+    # A list of all the unbounded PCollections in the pipeline. These unbounded
+    # pcollections will be cached.
+    self._unbounded_pcolls = unbounded_pcolls(self._unbounded_sources)
+
     # A mapping from PCollection id to python id() value in user defined
     # pipeline instance.
     (self._pcoll_version_map,
      self._cacheables,
      # A dict from pcoll_id to variable name of the referenced PCollection.
      # (Dict[str, str])
-     self._cacheable_var_by_pcoll_id) = cacheables(self.pcolls_to_pcoll_id)
+     self._cacheable_var_by_pcoll_id) = cacheables(self.pcolls_to_pcoll_id,
+                                                   self._unbounded_pcolls)
```

Comment:
Hi Sam, the `cacheables` are analyzed from the static global scope, so all PCollections should have already been included. There is no need to find and include them here.

Reply:
Ack, removed and added an ib.watch to the unit test.